micromamba/tests/test_env.py

```diff
@@ -1,152 +1,103 @@
-        assert str(env_3_fp) in env_json["envs"]
-
-        shutil.rmtree(env_2_fp)
-        env_json = run_env("list", "--json")
-        assert env_2_fp not in env_json["envs"]
-        assert env_3_fp in env_json["envs"]
-
-    def test_env_export(self):
-        env_name = "env-create-export"
-        spec_file = Path(__file__).parent / "env-create-export.yaml"
-        create("", "-n", env_name, "-f", spec_file)
-        ret = yaml.safe_load(run_env("export", "-n", env_name))
-        assert ret["name"] == env_name
-        assert set(ret["channels"]) == {"conda-forge"}
-        assert "micromamba=0.24.0=0" in ret["dependencies"]
-
-    def test_create(self):
-        # Tests for 'micromamba env create' can be found in 'test_create.py' (look for 'create_cmd')
-        pass
-
-    def test_env_remove(self):
-        env_name = "env-create-remove"
-        env_fp = str(self.root_prefix / "envs" / env_name)
-        conda_env_file = Path(os.path.join("~", ".conda/environments.txt")).expanduser()
-
-        # Create env with xtensor
-        res = create("xtensor", "-n", env_name, "--json", no_dry_run=True)
-
-        env_json = run_env("list", "--json")
-        assert env_fp in env_json["envs"]
-        assert Path(env_fp).expanduser().exists()
-        with open(conda_env_file, "r", encoding="utf-8") as f:
-            lines = [line.strip() for line in f]
-            assert env_fp in lines
-
-        # Unregister / remove env_name
-        run_env("remove", "-n", env_name, "-y")
-        env_json = run_env("list", "--json")
-        assert env_fp not in env_json["envs"]
-        assert not Path(env_fp).expanduser().exists()
-        with open(conda_env_file, "r", encoding="utf-8") as f:
-            lines = [line.strip() for line in f]
-            assert env_fp not in lines
+from . import helpers
+
+__this_dir__ = Path(__file__).parent.resolve()
+
+
+def test_env_list(tmp_home, tmp_root_prefix, tmp_empty_env):
+    env_json = helpers.run_env("list", "--json")
+
+    assert "envs" in env_json
+    assert len(env_json["envs"]) >= 2
+    assert str(tmp_root_prefix) in env_json["envs"]
+    assert str(tmp_empty_env) in env_json["envs"]
+
+
+def test_env_list_table(tmp_home, tmp_root_prefix, tmp_prefix):
+    res = helpers.run_env("list")
+
+    assert "Name" in res
+    assert "base" in res
+    assert str(tmp_root_prefix) in res
+    all_lines = res.splitlines()
+    print("\n".join(all_lines))
+    for line in all_lines:
+        if "*" in line:
+            active_env_l = line
+    assert str(tmp_root_prefix) in active_env_l
+
+    os.environ["CONDA_PREFIX"] = str(tmp_prefix)
+
+    res = helpers.run_env("list")
+
+    all_lines = res.splitlines()
+    for line in all_lines:
+        if "*" in line:
+            active_env_l = line
+    assert str(tmp_prefix) in active_env_l
+
+
+def test_register_new_env(tmp_home, tmp_root_prefix):
+    helpers.create("-n", "env2", "--json", no_dry_run=True)
```

Comment:
Do we remove the created test environments after the tests are done? It was handled by the `teardown_class` before... (This applies wherever `create` is used)

Reply:
This is handled by the `tmp_root_prefix` fixture that sets a temporary root prefix for all environments.
It is created under temporary storage, independent for each test, and safely cleaned up afterwards.

Reply:
Hmm yes, but it's not the case for the explicitly created envs `env2`, `env3`, etc.  Is it?

Reply:
Oh ok I see, you mean that deleting the `root_prefix` deletes implicitely everything... ok!